react/players/HTML5Player.tsx

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable jsx-a11y/media-has-caption */
+import React, { useRef } from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+
+import useVideo from '../useVideo'
+import TrackControl from '../controls/TrackControl'
+import styles from '../styles/styles.css'
+import VolumeControl from '../controls/VolumeControl'
+import PlayButton from '../controls/PlayButton'
+import FullscreenButton from '../controls/FullscreenButton'
+import { VideoPlayer } from '../interfaces'
+
+const CSS_HANDLES = [
+  'videoContainer',
+  'videoElement',
+  'controlsContainer',
+  'playButton',
+  'trackContainer',
+  'trackTimer',
+  'trackBar',
+  'fullscreenButton',
+  'volumeContainer',
+  'volumeSlider',
+  'volumeButton',
+]
+
+const FALLBACK_IMAGE_URL =
+  'https://storecomponents.vtexassets.com/arquivos/ids/155639'
+
+const HTML5Player: StorefrontFunctionComponent<VideoPlayer> = ({
+  src,
+  type,
+  width,
+  height,
+  autoPlay,
+  loop,
+  poster,
+  controlsType,
+  description,
+  muted,
+}) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const videoRef = useRef<HTMLVideoElement>(null)
+  const containerRef = useRef<HTMLDivElement>(null)
+  const {
+    isPlaying,
+    isMuted,
+    networkStatus,
+    play,
+    pause,
+    setVolume,
+    toggleMute,
+    toggleFullscreenMode,
+    changeState,
+  } = useVideo(videoRef, containerRef)
+
+  const hasCustomControls = controlsType === 'custom-vtex'
+  const hasNativeControls = controlsType === 'native'
+
+  const renderFallback = () => {
```

Comment:
Separate component

Reply:
Will it be better separate component rather than saving some shared code? @klzns